sql/core/src/main/scala/org/apache/spark/sql/parquet/newParquet.scala

```diff
@@ -361,13 +355,15 @@ private[sql] class ParquetRelation2(
         rawFooters.map(footer => footer.getFile -> footer).toMap
       }
 
+      footers = new RelationMemo(getFooters())
+
       // If we already get the schema, don't need to re-compute it since the schema merging is
       // time-consuming.
       if (dataSchema == null) {
         dataSchema = {
           val dataSchema0 = maybeDataSchema
-            .orElse(readSchema())
```

Comment:
In Metastore we trust. If the Metastore schema is available we bypass reading the footers.